src/endpoints/getCredential.ts

```diff
@@ -5,83 +5,94 @@ import jwt from 'jsonwebtoken';
 import { insertValuesIntoHandlebarsJsonTemplate } from '../helpers/handlebarhelpers';
 
 const secret =
-    process.env.PAYLOAD_SECRET ??
-    'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaabaaaaaaaaaaaaaaaaaaaa';
+  process.env.PAYLOAD_SECRET ??
+  'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaabaaaaaaaaaaaaaaaaaaaa';
 const expires = process.env.TOKEN_EXPIRATION_TIME_IN_SECONDS;
 
 export const getCredential: PayloadHandler = async (req, res) => {
-    let id: string;
+  let id: string;
 
-    const authHeader = req.headers.authorization;
+  const authHeader = req.headers.authorization;
 
-    try {
-        if (!authHeader.startsWith('Bearer ')) return res.sendStatus(401);
+  try {
+    if (!authHeader.startsWith('Bearer ')) return res.sendStatus(401);
 
-        const token = authHeader.split('Bearer ')[1];
+    const token = authHeader.split('Bearer ')[1];
 
-        const decoded = jwt.verify(token, secret);
+    const decoded = jwt.verify(token, secret);
 
-        if (typeof decoded === 'string' || !decoded.id) return res.sendStatus(401);
+    if (typeof decoded === 'string' || !decoded.id) return res.sendStatus(401);
 
-        id = decoded.id;
-    } catch (error) {
-        return res.sendStatus(401);
-    }
-
-    try {
-        const credential = await payload.findByID({
-            id,
-            collection: 'credential',
-            depth: 3,
-        });
+    id = decoded.id;
+  } catch (error) {
+    return res.sendStatus(401);
+  }
 
-        if (credential.status === 'REVOKED') {
-            return res.status(410).json({ reason: 'Credential has been revoked' });
-        }
+  try {
+    const credential = await payload.findByID({
+      id,
+      collection: 'credential',
+      depth: 3,
+    });
 
-        if (
-            typeof credential?.batch === 'string' ||
-            typeof credential.batch.template === 'string' ||
-            !credential.batch.template.credentialTemplateJson
-        ) {
-            return res.sendStatus(404);
-        }
+    if (credential.status === 'REVOKED') {
```

Comment:
Can we consider adding another check for tenant-match? Ideally it would not be possible to get credential data for TenantA by requesting the service via the TenantB domain.